djangocms_audio/models.py

```diff
@@ -0,0 +1,170 @@
+
+from cms.models import CMSPlugin
+
+from djangocms_attributes_field.fields import AttributesField
+
+from filer.fields.file import FilerFileField
+from filer.fields.folder import FilerFolderField
+
+
+"""
+Enables the user to add an "Audio player" plugin that serves as
+a wrapper rendering the player and its options.
+
+The "Audio player" plugin allows to add either a single "File" or a reference
+to a "Folder" as children.
+"""
+
+
+# mp3 is supported by all major browsers
+ALLOWED_EXTENSIONS = getattr(
+    settings,
+    'DJANGOCMS_AUDIO_ALLOWED_EXTENSIONS',
+    ['mp3', 'ogg'],
+)
+
+
+# Add additional choices through the ``settings.py``.
+def get_templates():
+    choices = getattr(
+        settings,
+        'DJANGOCMS_AUDIO_TEMPLATES',
+        False,
+    )
+    if choices:
+        return choices
+    return []
+
+
+@python_2_unicode_compatible
+class AudioPlayer(CMSPlugin):
+    """
+    Renders a container around the HTML <audio> elements.
+    """
+    DEFAULT_CHOICE = 'default'
+
+    TEMPLATE_CHOICES = [
+        ('default', _('Default')),
+    ]
+
+    # The label will be displayed as help text in the structure board view.
+    template = models.CharField(
+        verbose_name=_('Template'),
+        choices=TEMPLATE_CHOICES + get_templates(),
+        default=DEFAULT_CHOICE,
+        max_length=50,
+    )
+    label = models.CharField(
+        verbose_name=_('Label'),
+        blank=True,
+        max_length=200,
+    )
+    is_active = models.BooleanField(
+        verbose_name=_('Is active'),
+        default=True,
+    )
+    attributes = AttributesField(
+        verbose_name=_('Attributes'),
+        blank=True,
+    )
+
+    def __str__(self):
+        return self.label
+
+
+@python_2_unicode_compatible
+class AudioFile(CMSPlugin):
+    """
+    Renders the HTML <audio> element, add params through attributes.
+    """
+    audio_file = FilerFileField(
+        verbose_name=_('File'),
+        blank=False,
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name='+',
+    )
+    text_title = models.CharField(
+        verbose_name=_('Title'),
+        blank=True,
+        max_length=200,
+    )
+    text_description = models.TextField(
+        verbose_name=_('Description'),
+        blank=True,
+    )
+    text_transcript = models.TextField(
+        verbose_name=_('Text transcript'),
+        blank=True,
+        help_text=_('Provide a text transcript for accessibility support.'),
+    )
+    attributes = AttributesField(
+        verbose_name=_('Attributes'),
+        blank=True,
+    )
+
+    def __str__(self):
+        if self.audio_file_id and self.audio_file.label:
+            return self.audio_file.label
+        return '%s' % self.pk
+
+    def clean(self):
+        if (self.audio_file and
+            self.audio_file.extension not in ALLOWED_EXTENSIONS):
+            raise ValidationError(
+                ugettext('Incorrect file type: %s.') %
```

Comment:
don't use normal string formatting here, instead use a named variable, otherwise is not easy for translators to know what this string is.

```python
ugettext('Incorrect file type: %(extension)s.') % {'extension': self.audio_file.extension}
```